test/ok/error_test.exs

```diff
@@ -0,0 +1,35 @@
+defmodule OK.ErrorTest do
+  use ExUnit.Case
+
+  def foo(x) do
+    {:ok, x}
+  end
+
+  def bar(y) do
+    {:bad, y}
+  end
+
+  require OK
+
+  test "helpful error for binding failure" do
+    message = """
+    Binding to variable failed, '{:bad, 6}' is not a result tuple.
+
+        Code
+          b <- bar(a)
+
+        Expected signature
+          bar(a) :: {:ok, b} | {:error, reason}
+
+        Actual values
+          bar(a) :: {:bad, 6}
+    """
+    assert_raise OK.BindError, message, fn() ->
+      OK.with do
+        a <- foo(6)
+        b <- bar(a)
+        _ <- {:ok, b}
```

Comment:
1. This compiler error was always there. it is the same situation as you fixed earlier but in a slightly different place because you fixed it when the last line was not a bind.

Most people would probably write
```elixir
a <- foo(6)
b <- bar(a)
{:ok, b}
```
But maybe it is worth fixing.

2. This case clause error is from the case clause at line 221. It's technically not a bind error, there is no bind in that line. It is a return value error because with requires you to return a result as the last line. I was tempted to lift this restriction.